x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptorTests.java

```diff
@@ -1225,234 +1344,8 @@ public void testHasPrivilegesOtherThanIndex() {
         assertThat(roleDescriptor.hasUnsupportedPrivilegesInsideAPIKeyConnectedRemoteCluster(), equalTo(expected));
     }
 
-    public static List<RoleDescriptor> randomUniquelyNamedRoleDescriptors(int minSize, int maxSize) {
-        return randomValueOtherThanMany(
-            roleDescriptors -> roleDescriptors.stream().map(RoleDescriptor::getName).distinct().count() != roleDescriptors.size(),
-            () -> randomList(minSize, maxSize, () -> randomRoleDescriptor(false))
-        );
-    }
-
-    public static RoleDescriptor randomRoleDescriptor() {
-        return randomRoleDescriptor(true);
-    }
-
-    public static RoleDescriptor randomRoleDescriptor(boolean allowReservedMetadata) {
-        return randomRoleDescriptor(allowReservedMetadata, false, false, false);
-    }
-
-    public static RoleDescriptor randomRoleDescriptor(
-        boolean allowReservedMetadata,
-        boolean allowRemoteIndices,
-        boolean allowWorkflows,
-        boolean allowRemoteClusters
-    ) {
-        final RoleDescriptor.RemoteIndicesPrivileges[] remoteIndexPrivileges;
-        if (false == allowRemoteIndices || randomBoolean()) {
-            remoteIndexPrivileges = null;
-        } else {
-            remoteIndexPrivileges = randomRemoteIndicesPrivileges(0, 3);
-        }
-
-        RemoteClusterPermissions remoteClusters = RemoteClusterPermissions.NONE;
-        if (allowRemoteClusters && randomBoolean()) {
-            randomRemoteClusterPermissions(randomIntBetween(1, 5));
```

Comment:
While refactoring random role descriptor creation, I've noticed that we never assigned `randomRemoteClusterPermissions ` to `remoteClusters `. This is fixed in `RoleDescriptorTestHelper`.

Reply:
whoops ! thanks for fixing.